mobile/reducers/tables-data/tables_data_reducer.js

```diff
@@ -14,7 +16,30 @@ export default function (state = initialTablesData, action) {
         ...state,
         inchargesList: action.payload || [],
       };
-
+    case CREATE_HEALTH_WORKER_REQUEST: {
+      const newChwList = _.clone(state.chwList);
+      const { newHealthWorker } = action.payload;
+      newHealthWorker.needSynchronize = true;
+      newChwList.push(newHealthWorker);
+      return {
+        ...state,
+        chwList: newChwList,
+      };
+    }
+    case CREATE_HEALTH_WORKER_SUCCESS: {
+      const { id } = action.meta;
+      const newChwList = state.chwList.map((chw) => {
+        const newChw = chw;
+        if (chw.chwId === id) {
+          newChw.needSynchronize = false;
```

Comment:
yous should not mutate the state, return a new chw object here {...chw, needSynchronize: false}